packages/enterprise/src/license.ts

```diff
@@ -461,18 +473,55 @@ export function resetInMemoryLicenseCache(): void {
   });
 }
 
+function shouldLimitAccess(orgId: string): boolean {
+  if (shouldLimitAccessDueToExpiredLicense()) {
+    return true;
+  }
+
+  if (
+    orgId &&
+    licenseData?.organizationId &&
+    orgId !== licenseData.organizationId
+  ) {
+    return true;
+  }
+
+  if (licenseData?.remoteDowngrade) {
+    return true;
+  }
+
+  return false;
+}
+
+export function getEffectiveAccountPlan(org: MinimalOrganization): AccountPlan {
+  if (process.env.IS_CLOUD) {
+    return getAccountPlan(org);
+  }
+  const hasError = shouldLimitAccess(org.id);
+  const basicPlan = "oss";
```

Comment:
@james How quickly could someone's plan get into a state where `hasError === true`? In the past, we've been very lenient, especially with high paying enterprise accounts where if there is an issue, especially a billing issue, we don't disable their account functionality immediately, and instead allow them to continue operating and reach out to the org directly.

I think I'm seeing that we control when an org is remotely downgraded, as opposed to it being done programatically. Is that correct?

Just want to make sure we stay consistent (unless this has already been discussed).

CC: @jdorn

Reply:
Correct.  hasError will be true if:

1. A trial license has expired
2. A license has the wrong orgId
3. A remotelyDowngraded flag gets set manually by someone from Retool.

Hence a paying customer with a working license, will continue to have all functionality until someone manually says enough is enough and turns them off.